container/test-impl-base/src/test/java/org/jboss/arquillian/container/test/impl/client/deployment/AutomaticDeploymentScenarioGeneratorTestCase.java

```diff
@@ -0,0 +1,97 @@
+package org.jboss.arquillian.container.test.impl.client.deployment;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.jboss.arquillian.container.spi.client.deployment.DeploymentDescription;
+import org.jboss.arquillian.container.test.api.BeforeDeployment;
+import org.jboss.arquillian.container.test.api.DeploymentContent;
+import org.jboss.arquillian.container.test.spi.client.deployment.AutomaticDeployment;
+import org.jboss.arquillian.container.test.spi.util.ServiceLoader;
+import org.jboss.arquillian.test.spi.TestClass;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AutomaticDeploymentScenarioGeneratorTestCase {
```

Comment:
Just wondering - we might want to have some functional test for that too - is it something you would introcude in Chameleon?

Reply:
Yes. I think that this is like there is no ftest in Arquillian core, in the sense of starting a container and doing a deployment.

Reply:
Yeah, but I have a feeling we need to solve it to have build and ftests running together. Something to think about

Reply:
What I think is that we need to trigger chameleon build as a result of core success, so if both are green then we know that everything is good. In chameleon we have container tests and deployment tests.